### PR TITLE
Add specification language for name validity.

### LINKED ISSUE #53: Specify some basic requirements on the format of handler and app names.
The engine implementation may place its own requirements on the structure of names, but I think we should place some basic requirements in the dogma "spec" itself, something fairly basic like

- no whitespace 
- non-empty

I don't want to preclude using UTF-8 names or anything like that, but this at least sets a baseline for interoperability.

### PR DESCRIPTION
**Pull request by jmalloc**
Fixes #53

### CODE REVIEW

File: aggregate.go [hunk 1]
```diff
@@ -98,7 +98,12 @@ type AggregateConfigurer interface {
 	//
 	// It MUST be called exactly once within a single call to Configure().
 	//
-	// Each handler within an application MUST have a unique, non-empty name.
+	// The name MUST be a non-empty UTF-8 string consisting solely of printable
+	// unicode characters. A printable character is any character from the
```
Comment (danilvpetrov) [inline aggregate.go]: I wonder if `unicode` should start with the capital U as a standard name.

File: application.go [hunk 2]
```diff
@@ -18,6 +18,15 @@ type Application interface {
 // engine.
 type ApplicationConfigurer interface {
 	// Name sets the name of the application.
+	//
+	// It MUST be called exactly once within a single call to Configure().
+	//
+	// The name MUST be a non-empty UTF-8 string consisting solely of printable
+	// unicode characters. A printable character is any character from the
```
Comment (danilvpetrov) [inline application.go]: same as above.

File: integration.go [hunk 3]
```diff
@@ -51,7 +51,12 @@ type IntegrationConfigurer interface {
 	//
 	// It MUST be called exactly once within a single call to Configure().
 	//
-	// Each handler within an application MUST have a unique, non-empty name.
+	// The name MUST be a non-empty UTF-8 string consisting solely of printable
+	// unicode characters. A printable character is any character from the
```
Comment (danilvpetrov) [inline integration.go]: same as the above.

File: process.go [hunk 4]
```diff
@@ -114,7 +114,12 @@ type ProcessConfigurer interface {
 	//
 	// It MUST be called exactly once within a single call to Configure().
 	//
-	// Each handler within an application MUST have a unique, non-empty name.
+	// The name MUST be a non-empty UTF-8 string consisting solely of printable
+	// unicode characters. A printable character is any character from the
```
Comment (danilvpetrov) [inline process.go]: same as above

File: projection.go [hunk 5]
```diff
@@ -60,7 +60,12 @@ type ProjectionConfigurer interface {
 	//
 	// It MUST be called exactly once within a single call to Configure().
 	//
-	// Each handler within an application MUST have a unique, non-empty name.
+	// The name MUST be a non-empty UTF-8 string consisting solely of printable
+	// unicode characters. A printable character is any character from the
```
Comment (danilvpetrov) [inline projection.go]: same as above.

File: aggregate.go (hunk 1 above)
Comment (koden-km) [reply aggregate.go]: I agree, it does seem to be capitalised everywhere its used on the web, including in Go lang docs.

Comment (jmalloc) [reply aggregate.go]: Yep, seems to be the case.

**Comment by jmalloc** [comment]
Updated docs to exclude whitespace from valid names.

**Pull request merged**